client/test/src/store/index.test.js

```diff
@@ -0,0 +1,392 @@
+          });
+        actions.getDailyData({ commit, dispatch });
+        expect(dispatch).toHaveBeenCalledWith('retrieveDailyData');
+      });
+    });
+    describe('retrieveDailyData', () => {
+      let data;
+      beforeEach(() => {
+        process.env.VUE_APP_API_URL = 'app-api-url';
+        state.selectedLanguages = ['japanese', 'korean'];
+        data = {
+          created_at: new Date(),
+          dailyData: [],
+        };
+      });
+      it('calls the get endpoint for the selected languages', () => {
+        axios.get.mockResolvedValue({ data });
+        actions.retrieveDailyData({ commit, dispatch });
+        expect(axios.get).toHaveBeenCalledWith(
+          'app-api-url/daily?languages=japanese,korean',
+        );
+      });
+      it('calls chrome storage set', done => {
+        axios.get.mockResolvedValue({ data });
+        actions.retrieveDailyData({ commit, dispatch });
+        axios
+          .get('app-api-url/daily?languages=japanese,korean')
+          .then(response => {
+            expect(set).toHaveBeenCalledWith({ dailyData: data });
+            done();
+          });
+      });
+      it('calls dispatch save daily data', done => {
+        axios.get.mockResolvedValue({ data });
+        actions.retrieveDailyData({ commit, dispatch });
+        axios
+          .get('app-api-url/daily?languages=japanese,korean')
+          .then(response => {
+            expect(dispatch).toHaveBeenCalledWith('saveDailyData', data);
+            done();
+          });
+      });
+      it('calls commit set loading', done => {
+        axios.get.mockResolvedValue({ data });
+        actions.retrieveDailyData({ commit, dispatch });
+        axios
+          .get('app-api-url/daily?languages=japanese,korean')
+          .then(response => {
+            expect(commit).toHaveBeenCalledWith('setLoading', false);
+            done();
+          });
+      });
+      it('calls set has error if the call fails', done => {
+        axios.get.mockRejectedValue(new Error('Something bad happened!'));
+        actions.retrieveDailyData({ commit, dispatch });
+        axios
+          .get('app-api-url/daily?languages=japanese,korean')
+          .then(response => {
+            // empty then with catch tagged on to test the code inside the catch block
+          })
+          .catch(error => {
+            expect(commit).toHaveBeenCalledWith('setHasError', true);
+            done();
+          });
+      });
+    });
+    describe('saveDailyData', () => {
+      let data;
+      beforeEach(() => {
+        data = {
+          created_at: new Date(),
+          dailyData: [],
+        };
+      });
+      it('commits set daily data', () => {
+        actions.saveDailyData({ commit }, data);
+        expect(commit).toHaveBeenCalledWith('setDailyData', data);
+      });
+    });
+    describe('getLanguageOptions', () => {
+      let languageOptions;
+      beforeEach(() => {
+        languageOptions = [
+          'japanese',
```

Comment:
Is there a reason why this is written 11 times? :)

Reply:
At the time I did that so I could test the if statement here: 
` if (!!data && data.length === currentLanguagesCount)`
I probably could have made that more obvious by not using the same language 11 times though. :sweat_smile: Anyway I adjusted it to set the length of the array to be currentLanguagesCount in the test. This way too as more languages get added the test should not need to be updated.

Reply:
Got this!